IntegrationTest/DashboardEndpointsTest.cs

```diff
@@ -0,0 +1,420 @@
+﻿using IntegrationTest.Setup;
+using API;
+using Core.Exercises.Contracts;
+using Core.Exercises.Models;
+using Core.Languages.Models;
+using Core.Shared;
+using Core.Solutions.Contracts;
+using Core.Solutions.Models;
+using FluentResults;
+using Microsoft.Extensions.DependencyInjection;
+using NSubstitute;
+using System.Net.Http.Json;
+using System.Net;
+using Core.Sessions.Contracts;
+using Core.Sessions.Models;
+using Core.Dashboards.Contracts;
+using Core.Dashboards.Models;
+using System.Text.Json;
+
+namespace IntegrationTest;
+
+[Collection(CollectionDefinitions.NonParallelCollectionName)]
+public class DashboardEndpointsTest : IClassFixture<TestWebApplicationFactory<Program>>
+{
+    private readonly HttpClient _client;
+    private readonly TestWebApplicationFactory<Program> _factory;
+
+    public DashboardEndpointsTest(TestWebApplicationFactory<Program> factory)
+    {
+        _factory = factory;
+        _client = factory.CreateClient();
+    }
+
+    [Fact]
+    public async Task GetExercisesInSession_ShouldReturn_FailedAuthentication()
```

Comment:
Jeg vil være fan af vi skriver grunden til fejlen ind i testcases, så I dette tilfælde vil jeg tænke 
`GetExercisesInSession_AnonymousUser_ShouldReturn_...`

Det gælder egentlig for de fleste test cases jeg lige har set her. Specielt nogle af de senere bliver det svært at gennemskue hvad problemet er, når det titlen ikke specifiere mere end `ShouldFail`

Reply:
Jeg har forsøgt at give mere deskriptive navne nu